lib/composables/useKFloatingInteraction.js

```diff
@@ -0,0 +1,468 @@
+import { ref, onMounted, onUnmounted, computed, nextTick } from 'vue';
+import { isNuxtServerSideRendering } from '../utils';
+import globalThemeState from '../styles/globalThemeState';
+
+// Attribute to be applied on a trigger element
+// Its value is ID of the associated floating element
+const ATTR_FLOATING_ID = 'data-floating-id';
+
+// TODO Add click
+// TODO Look into 'touchstart' browser support
+const SUPPORTED_INTERACTIONS = ['hover', 'touch', 'focus', 'keyboardfocus'];
+const DEFAULT_INTERACTIONS = ['hover'];
+
+const MOUSEENTER = 'mouseenter';
+const MOUSELEAVE = 'mouseleave';
+const TOUCHSTART = 'touchstart';
+const TOUCHEND = 'touchend';
+const FOCUS = 'focus';
+const BLUR = 'blur';
+
+const INTERACTIONS_TO_EVENTS = {
+  hover: MOUSEENTER,
+  touch: TOUCHSTART,
+  focus: FOCUS,
+  keyboardfocus: FOCUS,
+};
+
+const EVENT_HANDLERS = {
+  [MOUSEENTER]: onMouseEnter,
+  [TOUCHSTART]: onTouchStart,
+  [FOCUS]: onFocus,
+};
+
+// If 'delegateTo' is 'root', this map is used to decide
+// whether an event should be attached to Document or Window
+const DELEGATE_ROOT = 'root';
+const DELEGATE_ROOT_TARGET = {
+  [MOUSEENTER]: 'document',
+  [TOUCHSTART]: 'document',
+  [FOCUS]: 'window',
+};
+
+// For each mounted floating element stores
+// configured interactions
+// { <floating element id>: { activateOn: [ ... ] } }
+export const _floatingInteractions = {};
+
+// For each mounted floating element stores elements cache
+// for its trigger (and delegate if provided) element:
+// { <floating element id> : { <selector>: <element> } }
+// Used to avoid repeated DOM queries.
+export const _floatingCache = {};
+
+// For each delegate element stores types of events
+// attached to it and the number of related mounted
+// floating elements depending on each event type:
+// { <delegate element id>: { <event type> : <mounted floating elements count> } } }
+// Used to determine when to remove listeners from delegates.
+export const _delegateUsage = {};
+
+// Currently active floating elements and their
+// trigger elements { <floating element id>: <trigger element> }
+const activeFloatingElements = ref({});
+
+function getFloatingId(triggerEl) {
+  return triggerEl?.dataset?.floatingId;
+}
+
+function areInteractionsValid(interactions) {
+  return interactions.every(i => SUPPORTED_INTERACTIONS.includes(i));
```

Comment:
Maybe check that interactions is even an Array first?

Not a huge deal, but I suppose SUPPORTED_INTERACTIONS could be a set, and we could just check that a set of interactions is a subset. Looks like `isSubsetOf` is polyfillable by core.js so we may be able to use it directly: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/isSubsetOf